xcube_cds/datasets/satellite_soil_moisture.py

```diff
@@ -119,6 +119,14 @@ def read_file(self, dataset_name: str, cds_api_params: Dict,
         # and inspection of some downloaded files confirms it.
         ds = xr.open_mfdataset(paths, combine='by_coords')
         ds.attrs.update(self.combine_netcdf_time_limits(paths))
+
+        # Subsetting is not supported by the soil moisture dataset, so we
+        # handle it here instead.
+        if "bbox" in open_params and \
+                open_params["bbox"] != [-180, -90, 180, 90]:
+            xmin, ymin, xmax, ymax = open_params["bbox"]
+            ds = ds.sel(lon=slice(xmin, xmax), lat=slice(ymax, ymin))
```

Comment:
I fear, this code does not handle bboxes correctly whose longitude cross the anti-meridian.